### PR TITLE
Setup: Add links to Artifact section

### PR DESCRIPTION
**Pull request by fifthposition**
Also add a Setup page on setting up Maven artifact.

### CODE REVIEW

**Comment by fifthposition** [comment]
Partially addresses #1613.

File: _data/navigation.yml [hunk 1]
```diff
@@ -114,20 +114,22 @@ setup:
...
             url: /setup/artifacts/gitlab/
-          - title: S3
-            url: /setup/artifacts/s3/
+          - title: HTTP
+            url: /setup/artifacts/http/
+          - title: Maven
+            url: /setup/artifacts/maven/
           - title: Oracle Object
             url: /setup/artifacts/oracle/
+          - title: Amazon S3
```
Comment (dorbin) [inline _data/navigation.yml]: Maybe not important, but arguably this would go up above Bitbucket, a-for-Amazon rather than s for s3.

**Review by dorbin — APPROVED** [review]
LGTM, Ben, thanks!

**Comment by dorbin** [comment]
@plumpy, fyi

File: _data/navigation.yml (hunk 1 above)
Comment (fifthposition) [reply _data/navigation.yml]: Fixed. Thanks!

File: setup/artifacts/index.md [hunk 2]
```diff
@@ -12,7 +12,11 @@ their ["type"](/reference/artifacts), they will need credentials to be
 downloaded or uploaded. These pages take you through configuring those
 credentials.
 
+* [Bitbucket](/setup/artifacts/bitbucket/)
```
Comment (plumpy) [inline setup/artifacts/index.md:16]: These are almost alphabetical, but not quite (Google goes down, Amazon comes up), might as well sort them?

Comment (fifthposition) [reply setup/artifacts/index.md:16]: Huh. I really thought I did... guess I need to go back to work on my alphabet some more 🤦‍♂ Will fix!

Comment (fifthposition) [reply setup/artifacts/index.md:16]: Oh... I may not have sorted the _page’s_ list. I did sort the nav. Never mind

**Pull request merged**